### DISCUSSION TITLE
Would using a CodeGen tool help automate the process of creating bindings for Jolt?

### DISCUSSION THREAD

**Discussion by PhilT** [amerkoleci/JoltPhysicsSharp#56] [Announcements]
The Silk.NET team are always banging on about how they use codegen tools to generate the bindings for a lot of their libraries. Could something similar be used here to help automate the process?

**Comment by aSemy** [comment]
Hi 👋. I've been trying to automatically generate C bindings for Jolt too. I've been approaching the problem by using the WebIDL file for [JoltPhysics.js](https://github.com/jrouwe/JoltPhysics.js), and then generating pure C binding code.

Would pure C bindings also be useful for C#?

I've been writing my bindings for Kotlin/Native, which has a tool for generating Kotlin bindings for any C code.

I'll tidy up my work so far, and make the repo public later this week. It would be nice to have a collaborator! I'm not very good at C++, so trying to figure out how to write C bindings is tough.


### Related 

* https://github.com/amerkoleci/JoltPhysicsSharp/discussions/25

**Reply by aSemy** [reply]
Here's my progress so far:

https://github.com/aSemy/joltc

They don't work! But I can see the finishing line.

**Comment by amerkoleci** [comment]
Hi 
I agree that using CodeGen would make bindings easier, I've did that in different projects although it's not an easy process, hand crafted bindings are more slow but more accurate, will investigate some code generation in the future. 

Cheers
